### PR TITLE
Move to fileglob loop for yml requirement locations

### PR DESCRIPTION
**Pull request by AlanCoding**
##### SUMMARY
Overdue followup on https://github.com/ansible/awx/pull/8312

I knew there was more research needed, but didn't know what form it would take until I got my hands dirty.

This entails a _look and feel_ change.

##### ISSUE TYPE
 - Feature Pull Request

##### COMPONENT NAME
 - API

##### ADDITIONAL INFORMATION
Screenshot from 2021-04-19 15-46-47

It has a warning... but I don't dislike it.

### CODE REVIEW

File: awx/playbooks/project_update.yml [hunk 1]
```diff
@@ -187,22 +182,19 @@
...
           args:
             chdir: "{{project_path|quote}}"
           register: galaxy_collection_result
-          with_items: "{{ doesCollectionRequirementsExist.results }}"
-          when: item.stat.exists
+          with_fileglob:
+            - "{{project_path|quote}}/collections/requirements.yaml"
+            - "{{project_path|quote}}/collections/requirements.yml"
+            - "{{project_path|quote}}/requirements.yaml"
+            - "{{project_path|quote}}/requirements.yml"
```
Comment (AlanCoding) [inline awx/playbooks/project_update.yml:193]: @geerlingguy with https://github.com/ansible/awx/issues/106 in mind, am I getting hotter or colder?

Did Ansible itself roll out a combined `ansible-galaxy install` command that handled both collections and roles? If it did, then using that for the top-level requirements would be preferable.

**Review by thedoubl3j — APPROVED** [review]
look good to me @AlanCoding bringing it down to 2 tasks will make future edits easier and create less bloat.

File: awx/playbooks/project_update.yml (hunk 1 above)
Comment (AlanCoding) [reply awx/playbooks/project_update.yml:193]: Was thinking about https://github.com/ansible/ansible/pull/67843 by @jborean93 

and after reading back over that, I believe there's nothing we can really use from it. We still support Ansible 2.9 (which doesn't have the combined command), and we have to specify the roles path, which isn't compatible with the combined roles & collections install command.

So I think this answers the question for me - we have to keep using `ansible-galaxy` `role`/`collection` `install` forever.

Now I'm questioning if we should install roles from `{{project_path|quote}}/requirements.yaml` or not.

**Comment by fourstepper** [comment]
Is this backwards compatible? I am having some issues with a project that has the requirements in the `roles/` directory

**Comment by AlanCoding** [comment]
yeah, it should be completely backward compatible. If you have some details we'll be interested.

**Comment by fourstepper** [comment]
I am getting the following warning when fetching the requirements, could it be related?:

https://paste.sr.ht/~fourstepper/8dd481a03cc7389350409efdef37f5908661cdcf

**Comment by AlanCoding** [comment]
No, that's expected. The aesthetics of it are a drawback of this method. All its saying is that you don't have the `collections/` folder, and it was in the list that was searched for. If you have stuff in `roles/`, that task isn't even relevant to you. There's a different task that is expected to install roles.

**Comment by fourstepper** [comment]
Yeah, my bad. Everything is working fine

**Pull request merged**